spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceTablesBase.java

```diff
@@ -2223,4 +2254,36 @@ private DeleteFile writeEqDeleteFile(Table table) {
       throw new RuntimeException(e);
     }
   }
+
+  private long sumDataFileSizeInBytes(Iterable<DataFile> dataFiles) {
+    return StreamSupport.stream(dataFiles.spliterator(), false)
+        .mapToLong(DataFile::fileSizeInBytes)
+        .sum();
+  }
+
+  private List<DataFile> listDataFilesFromCommitId(Table table, long commitId) {
+    return StreamSupport.stream(
+            table.snapshot(commitId).addedDataFiles(table.io()).spliterator(), false)
+        .collect(Collectors.toList());
+  }
+
+  private void assertPartitionIdFromDataFiles(
+      List<DataFile> dataFilesFromFirstCommit, List<DataFile> dataFilesFromSecondCommit) {
```

Comment:
I am thinking if we write the helper method this way, then it probably cannot be reused. I would suggest to put assertion expected value as parameters (like numberOfDataFiles and partitionValue), everything else looks good to me

Reply:
Good point! fixed now.